src/widget/form/chatform.cpp

```diff
@@ -784,39 +784,44 @@ void ChatForm::onScreenshotClicked()
 void ChatForm::doScreenshot()
 {
     ScreenshotGrabber* screenshotGrabber = new ScreenshotGrabber(this);
-    connect(screenshotGrabber, &ScreenshotGrabber::screenshotTaken, this, &ChatForm::onScreenshotTaken);
-    screenshotGrabber->showGrabber();
-    // Create dir for screenshots
-    QDir(Settings::getInstance().getAppDataDirPath()).mkpath("screenshots");
-}
+    connect(screenshotGrabber, &ScreenshotGrabber::screenshotTaken,
+            this, [screenshotGrabber, this](const QPixmap &pixmap)
+    {
+        // use ~ISO 8601 for screenshot timestamp, considering FS limitations
+        // https://en.wikipedia.org/wiki/ISO_8601
+        // Windows has to be supported, thus filename can't have `:` in it :/
+        // Format should be: `qTox_Screenshot_yyyy-MM-dd HH-mm-ss.zzz.png`
+        QString filepath = QString("%1screenshots%2qTox_Screenshot_%3.png")
+                               .arg(Settings::getInstance().getAppDataDirPath())
+                               .arg(QDir::separator())
+                               .arg(QDateTime::currentDateTime().toString("yyyy-MM-dd HH-mm-ss.zzz"));
+
+        QFile file(filepath);
+
+        if (!file.open(QFile::ReadWrite))
+        {
+            QMessageBox::warning(this,
+                                 tr("Failed to open temporary file", "Temporary file for screenshot"),
+                                 tr("qTox wasn't able to save the screenshot"));
 
-void ChatForm::onScreenshotTaken(const QPixmap &pixmap) {
-    // use ~ISO 8601 for screenshot timestamp, considering FS limitations
-    // https://en.wikipedia.org/wiki/ISO_8601
-    // Windows has to be supported, thus filename can't have `:` in it :/
-    // Format should be: `qTox_Screenshot_yyyy-MM-dd HH-mm-ss.zzz.png`
-    QString filepath = QString("%1screenshots%2qTox_Screenshot_%3.png")
-                           .arg(Settings::getInstance().getAppDataDirPath())
-                           .arg(QDir::separator())
-                           .arg(QDateTime::currentDateTime().toString("yyyy-MM-dd HH-mm-ss.zzz"));
+            delete screenshotGrabber;
+            return;
+        }
 
-    QFile file(filepath);
+        pixmap.save(&file, "PNG");
 
-    if (!file.open(QFile::ReadWrite))
-    {
-        QMessageBox::warning(this,
-                             tr("Failed to open temporary file", "Temporary file for screenshot"),
-                             tr("qTox wasn't able to save the screenshot"));
-        return;
-    }
+        long long filesize = file.size();
+        file.close();
+        QFileInfo fi(file);
 
-    pixmap.save(&file, "PNG");
+        emit sendFile(f->getFriendID(), fi.fileName(), fi.filePath(), filesize);
+        delete screenshotGrabber;
```

Comment:
Yes, but it's added as child object (otherwise could become a dangling pointer). Thus, exiting qTox from within screenshot grabber actually leads to SIGSEGV. ^^